### PR TITLE
[chore] - Use custom context for archive handler of specialized archives

### PR DESCRIPTION
**Pull request by ahrav**
### Description:
Provide the custom context with a logger to the HandleSpecialized method and use the reReader. It seems like the archive package isn't able to correctly determine the format for an io.MultiReader.

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/handlers/handlers.go [hunk 1]
```diff
@@ -43,15 +47,29 @@ func HandleFile(ctx context.Context, file io.Reader, chunkSkel *sources.Chunk, c
...
+		}
+		defer reReader.Close()
 		if specialHandler, ok := h.(SpecializedHandler); ok {
-			if file, isSpecial, err = specialHandler.HandleSpecialized(ctx, file); isSpecial && err == nil {
-				return handleChunks(ctx, h.FromFile(ctx, file), chunkSkel, chunksChan)
+			if file, isSpecial, err = specialHandler.HandleSpecialized(aCtx, reReader); isSpecial && err == nil {
+				return handleChunks(aCtx, h.FromFile(ctx, file), chunkSkel, chunksChan)
+			}
+			if err != nil {
+				aCtx.Logger().Error(err, "error processing specialized file")
```
[the remark this answers is not in the document]
Comment (ahrav) [reply pkg/handlers/handlers.go:61]: Yea good point, i forgot that Archive also includes non-specialized formats. Will fix.

[the remark this answers is not in the document]
Comment (ahrav) [reply pkg/handlers/handlers.go:62]: Actually makes the code a bit nicer by removing the vars and using `:=` instead. thanks

[the remark this answers is not in the document]
Comment (rosecodym) [reply pkg/handlers/handlers.go:62]: it does look cleaner but my original comment still technically applies :v. if there's an error here the log message will report that the failure was in processing a specialized file, even if `isSpecial == false`. do we know by this point that the file is guaranteed to be specialized, even if `isSpecial == false`?

[the remark this answers is not in the document]
Comment (ahrav) [reply pkg/handlers/handlers.go]: I modified the error message. Also added a comment that mentions if `isSpecial` is true there wont be an error.

**Review by rosecodym — APPROVED** [review]

**Pull request merged**